### ISSUE TITLE
Compiler quietly accepts verify() in the controls

### ISSUE THREAD

**Issue by vgurevich** [p4lang/p4c#608]
Hello,

Through my ignorance, I've written the following:
```
error { BadIPv4Checksum }

control MyVerifyChecksum(
    in    my_headers_t   hdr,
    inout my_metadata_t  meta)
{
    Checksum16() ipv4_checksum;
    bit<16>      checksum;
    
    apply {
        if (hdr.ipv4.isValid()) {
            checksum = ipv4_checksum.get(
                {    
                    hdr.ipv4.version,
                    hdr.ipv4.ihl,
                    hdr.ipv4.diffserv,
                    hdr.ipv4.totalLen,
                    hdr.ipv4.identification,
                    hdr.ipv4.flags,
                    hdr.ipv4.fragOffset,
                    hdr.ipv4.ttl,
                    hdr.ipv4.protocol,
                    hdr.ipv4.srcAddr,
                    hdr.ipv4.dstAddr
                });
            verify(hdr.ipv4.hdrChecksum == checksum, error.BadIPv4Checksum);
        }
    }
}
```

The compiler happily accepted the code and generated a JSON. 

The right thing to do would be to reject the program, since verify() can appear in parser states only as per spec.

**Comment by jnfoster** [comment]
Good catch. I pushed a possible fix.

**Comment by jnfoster** [comment]
Fixed by #611
